File: src/main/java/com/indico/mutation/ModelGroupPredict.java
package com.indico.mutation;

import java.util.List;
import com.apollographql.apollo.ApolloCall;
import com.apollographql.apollo.ApolloClient;
import com.apollographql.apollo.api.Response;

import com.indico.IndicoClient;
import com.indico.jobs.Job;
import com.indico.jobs.JobOptions;
import com.indico.Async;
import com.indico.Mutation;
import com.indico.entity.ModelGroup;
import com.indico.PredictModelGraphQLMutation;

public class ModelGroupPredict implements Mutation<Job> {

    private int modelId;
    private List<String> data;
    private JobOptions jobOptions;
    private final IndicoClient indicoClient;

    public ModelGroupPredict(IndicoClient indicoClient) {
        this.indicoClient = indicoClient;
    }

    /**
     * Use to predict ModelGroup
     * 
     * @param modelGroup ModelGroup
     * @return ModelGroupPredict
     */
    public ModelGroupPredict modelGroup(ModelGroup modelGroup) {
        modelId = modelGroup.selectedModel.id;
        return this;
    }

    /**
     * Use to predict ModelGroup by id
     * 
     * @param modelId Model id
     * @return ModelGroupPredict
     */
    public ModelGroupPredict modelId(int modelId) {
        this.modelId = modelId;
        return this;
    }

    /**
     * Data to predict
     * 
     * @param data Data
     * @return ModelGroupPredict
     */
    public ModelGroupPredict data(List<String> data) {
        this.data = data;
        return this;
    }

    /**
     * Job Options for Job
     * 
     * @param jobOptions Job options
     * @return ModeGroupPredict
     */
    public ModelGroupPredict jobOptions(JobOptions jobOptions) {
        this.jobOptions = jobOptions;
        return this;
    }

    /**
     * Executes request and returns job 
     * 
     * @return Job
     */
    @Override
    public Job execute() {
        ApolloCall<PredictModelGraphQLMutation.Data> apolloCall = this.indicoClient.apolloClient.mutate(PredictModelGraphQLMutation.builder()
                .modelId(modelId)
                .data(data)
                .build());
        Response<PredictModelGraphQLMutation.Data> response = (Response<PredictModelGraphQLMutation.Data>) Async.executeSync(apolloCall, this.indicoClient.config).join();
        String jobId = response.data().modelPredict().jobId();
        return new Job(this.indicoClient,jobId);
    }
}

File: src/main/java/com/indico/mutations/ModelGroupPredict.kt
package com.indico.mutations

import com.indico.IndicoKtorClient
import com.indico.entity.ModelGroup
import com.indico.exceptions.IndicoMutationException
import com.indico.graphql.PredictModelGraphQL
import com.indico.query.Job

class ModelGroupPredict(private val indicoClient: IndicoKtorClient) : Mutation<Job?, PredictModelGraphQL.Result>() {
    private var modelId = 0
    private var data: List<String?> = emptyList()

    /**
     * Use to predict ModelGroup
     *
     * @param modelGroup ModelGroup
     * @return ModelGroupPredict
     */
    fun modelGroup(modelGroup: ModelGroup): ModelGroupPredict {
        modelId = modelGroup.selectedModel.id
        return this
    }

    /**
     * Use to predict ModelGroup by id
     *
     * @param modelId Model id
     * @return ModelGroupPredict
     */
    fun modelId(modelId: Int): ModelGroupPredict {
        this.modelId = modelId
        return this
    }

    /**
     * Data to predict
     *
     * @param data Data
     * @return ModelGroupPredict
     */
    fun data(data: List<String?>): ModelGroupPredict {
        this.data = data
        return this
    }

    /**
     * Executes request and returns job
     *
     * @return Job
     */
    override fun execute(): Job? {
        return try {
            val call = PredictModelGraphQL(
                variables = PredictModelGraphQL.Variables(
                    modelId = this.modelId,
                    data = this.data
                )
            )

            val response = indicoClient.execute(call)
            val jobId: String = response.data?.modelPredict?.jobId?:
                throw IndicoMutationException("Error retrieving job id for model group predict")
            Job(indicoClient, id = jobId, errors = emptyList())
        } catch (ex: RuntimeException) {
            throw IndicoMutationException("Call for Model Group Predict failed", ex)
        }
    }

}